### DISCUSSION TITLE
Cascade Delete js

### DISCUSSION THREAD

**Discussion by stychu** [supabase/supabase#3075] [Questions] [answered]
Is it possible to perform cascade deletion from Supbase UI and also from withing javascript?? Or do I have to make some SQL ?

**Comment by dshukertjr** [comment, marked as answer]
Hi @stychu!

Whether table deletion cascades or not depends on the table definition, so as long as a column is set to `delete cascade`, your deletion will cascade no matter how you delete it, from Supabase UI, or from js client. 

In order to create a column with `delete cascade` turned on, it seems like you would have to write some SQL at the moment. There is an [open issue](https://github.com/supabase/supabase/issues/2518) to enable this setting from within Supabase dashboard, so possibly this might be added as a future, but for now, you can run an SQL something like this to create your table:

```sql
CREATE TABLE order_items (
    product_no integer REFERENCES products ON DELETE CASCADE,
    order_id integer REFERENCES orders ON DELETE CASCADE,
    quantity integer,
    PRIMARY KEY (product_no, order_id)
);
```

**Reply by stychu** [reply]
Hey @dshukertjr How I can do that? I have only Supabase in javascript. Created all tables via dashboard. It is possible to update them when already created? If so how to do so. I would appreciate some guidance

**Reply by soedirgo** [reply]
@stychu I don't think you can alter existing columns to cascade deletes from the table editor, but you can use the SQL editor:
```sql
ALTER TABLE order_items
DROP CONSTRAINT order_items_product_no_fkey,
ADD CONSTRAINT order_items_product_no_fkey
   FOREIGN KEY (product_no)
   REFERENCES products
   ON DELETE CASCADE,
DROP CONSTRAINT order_items_order_id_fkey,
ADD CONSTRAINT order_items_order_id_fkey
   FOREIGN KEY (order_id)
   REFERENCES orders
   ON DELETE CASCADE;
```

**Reply by stychu** [reply]
Awesome I was able to alter it. Thanks guys!
